Framework/DataHandling/src/SaveAscii2.cpp

```diff
@@ -399,24 +397,26 @@ SaveAscii2::stringListToVector(std::string &inputString) {
 void SaveAscii2::populateQMetaData() {
   std::vector<std::string> qValues;
   const auto nHist = m_ws->getNumberHistograms();
+  const auto &spectrumInfo = m_ws->spectrumInfo();
   for (size_t i = 0; i < nHist; i++) {
     const auto specNo = m_ws->getSpectrum(i).getSpectrumNo();
     const auto workspaceIndex = m_specToIndexMap[specNo];
-    const auto detector = m_ws->getDetector(workspaceIndex);
-    double twoTheta(0.0), efixed(0.0);
-    if (!detector->isMonitor()) {
-      twoTheta = 0.5 * m_ws->detectorTwoTheta(*detector);
+    double theta(0.0), efixed(0.0);
+    if (!spectrumInfo.isMonitor(workspaceIndex)) {
+      theta = 0.5 * spectrumInfo.twoTheta(workspaceIndex);
       try {
+        boost::shared_ptr<const Geometry::IDetector> detector(
+            &spectrumInfo.detector(workspaceIndex), NoDeleting());
         efixed = m_ws->getEFixed(detector);
       } catch (std::runtime_error) {
         throw;
       }
     } else {
-      twoTheta = 0.0;
+      theta = 0.0;
```

Comment:
Yeah right, I should also fix `UnitConversion::run`. The name in the signature is also wrong.